test/integration/form-data.spec.ts

```diff
@@ -0,0 +1,52 @@
+import { getLocal } from "../..";
+import { expect, File, fetch as utilFetch } from "../test-utils";
+
+// workaround to use real fetch in node v18 and later
+const fetch = (typeof globalThis.fetch === 'undefined') ? utilFetch : globalThis.fetch;
+
+describe("FormData", () => {
+    let server = getLocal();
+
+    beforeEach(() => server.start());
+    afterEach(() => server.stop());
+
+    it("should parse application/x-www-form-urlencoded", async () => {
+        await server.forPost("/mocked-endpoint").thenCallback(async (req) => {
+            const formData = await req.body.getFormData() ;
+            return {
+                statusCode: 200,
+                body: JSON.stringify(formData),
+            };
+        });
+
+        await expect(
+            fetch(server.urlFor("/mocked-endpoint"), {
+                method: "POST",
+                body: "id=123&id=456&id=789&order=desc"
+            })
+        ).to.responseText("{\"id\":[\"123\",\"456\",\"789\"],\"order\":\"desc\"}");
+    });
+
+    it("should parse multipart/form-data", async () => {
+        await server.forPost("/mocked-endpoint").thenCallback(async (req) => {
+            const formData = await req.body.getFormData() ;
+            return {
+                statusCode: 200,
+                body: JSON.stringify(formData),
+            };
+        });
+
+        const formData = new FormData();
```

Comment:
This fails in Node 14 & 16 because those don't have FormData (we could drop them now of course, but that's a major bump and not strictly necessary for this). There's a FormData polyfill already in test-utils, so you just need to import that here.

Reply:
It appears that padding `FormData` polyfill to `fetch` works differently from real ones.
```diff
{
-  "[object FormData]": ""
+  "id": [
+    "123"
+    "456"
+    "789"
+  ]
+  "order": "desc"
+  "readme": "file content"
}
```

The polyfill `fetch` encodes the polyfill `FormData` into `[object FormData]` and `text/plain`.
I think we need a workaround to change this behavior.

Reply:
Ah, it fails if users use their own polyfill `fetch` and `FormData`.
Perhaps `getFormData` should throw `new Error("getformData is supported only in Node.js 18+` or such, since Node.js reached EOL on 2023-09-11 🤔

Reply:
I don't think getFormData is the problem here though - it looks like the parsing code will work fine in Node 16 and older versions, right? It sounds like the the issue is just in the test code (on the client side).

If the polyfills aren't workable there, tbh I think that's fine, let's just only run this test in this in modern versions. You can add a version check and `skip()` to skip the test, just follow how we do that in other tests for an example:

https://github.com/httptoolkit/mockttp/blob/309b2e66a22f1dff80a5a852d5494590a1806999/test/integration/proxying/https-proxying.spec.ts#L239-L241

The version numbers for the various detected features are all defined in test-util.